include/net/nrf_cloud.h

```diff
@@ -774,6 +774,19 @@ int nrf_cloud_fota_fmfu_apply(void);
  */
 bool nrf_cloud_fota_is_type_modem(const enum nrf_cloud_fota_type type);
 
+/**
+ * @brief Function to determine if the specified FOTA type is enabled by the
+ *        configuration. Not to be used by REST-based FOTA implementations, as
+ *        CONFIG_NRF_CLOUD_FOTA is a requirement for the FOTA type to be
```

Comment:
it would be up to the user to determine if their application supports FOTA.  in general, the REST-based FOTA implementation is left up to the application...  so i'm not sure if making this function work for both makes sense?
suggestions?

Reply:
As long as we continue to treat REST services as an application responsibility to manage, this is fine.  But perhaps saying "Not to be used" is too strong, as it sounds bad/dangerous.

Maybe say "Applications using REST-based FOTA are responsible for determining what FOTA types they support. 
  Only applications which use other transports, such as MQTT, should use this function."

Reply:
Or "need to use this function"

Reply:
Could the function be modified to return sane result both when CLOUD_REST and CLOUD_MQTT are enabled?

Reply:
it could... but it isn't intended to be used by REST, see above.

Reply:
updated comment and moved into nrf_cloud_fota to remove any usage ambiguity.

Reply:
No worries, thanks for trying! Plus, you managed to do it outside of a function body, which is all I really care about anyways, so kudos for that.

I'm pretty sure Zephyr has a ternary operator macro defined somewhere that might let you do what you did as a one-liner, I'll see if I can dig it up

Reply:
Such macros do exist, but they are not particularly readable. What you've done maximizes readability, I think

https://docs.zephyrproject.org/latest/kernel/util/index.html in case you decide you want to look at it yourself

`IF_ENABLED` and `COND_CODE_1` would be the macros of interest if you decided to go that route. But I think using obscure macros may actually make things worse for a beginner, whereas standard c macros are pretty readable